From 4dfa241312d6cff0ff23e3a60af252b75319a2e0 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Wed, 6 Nov 2013 17:09:10 +0800
Subject: [PATCH 476/502] ENGR00215174 update to pass build for android 4.4.

structure sensor_t add two member variables.
property_get() function limit the input buffer size to
more than 92 byte.
android 4.4 delete ANDROID_LENS_INFO_SHADING_MAP property
in camera_medata_tag.
so, comment the initialization for that.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 libsensors/sensors.cpp           | 10 +++++-----
 mx6/libcamera2/CameraModule.cpp  |  4 +++-
 mx6/libcamera2/CameraUtil.h      |  2 +-
 mx6/libcamera2/MetadaManager.cpp |  4 ++--
 mx6/libcamera2/StreamAdapter.cpp |  2 +-
 5 files changed, 12 insertions(+), 10 deletions(-)

diff --git a/libsensors/sensors.cpp b/libsensors/sensors.cpp
index 4690d34..1c93231 100755
--- a/libsensors/sensors.cpp
+++ b/libsensors/sensors.cpp
@@ -69,23 +69,23 @@ static const struct sensor_t sSensorList[] = {
         { "Freescale 3-axis Accelerometer",
           "Freescale Semiconductor Inc.",
           1, SENSORS_ACCELERATION_HANDLE,
-          SENSOR_TYPE_ACCELEROMETER, RANGE_A, CONVERT_A, 0.30f, 20000, { } },
+          SENSOR_TYPE_ACCELEROMETER, RANGE_A, CONVERT_A, 0.30f, 20000, 0, 0, { } },
         { "Freescale 3-axis Magnetic field sensor",
           "Freescale Semiconductor Inc.",
           1, SENSORS_MAGNETIC_FIELD_HANDLE,
-          SENSOR_TYPE_MAGNETIC_FIELD, 1500.0f, CONVERT_M, 0.50f, 100000, { } },
+          SENSOR_TYPE_MAGNETIC_FIELD, 1500.0f, CONVERT_M, 0.50f, 100000, 0, 0, { } },
         { "Freescale Orientation sensor",
           "Freescale Semiconductor Inc.",
           1, SENSORS_ORIENTATION_HANDLE,
-          SENSOR_TYPE_ORIENTATION, 360.0f, CONVERT_O, 0.50f, 100000, { } },
+          SENSOR_TYPE_ORIENTATION, 360.0f, CONVERT_O, 0.50f, 100000, 0, 0, { } },
         { "MPL3115 Temperature sensor",
           "Freescale Semiconductor Inc.",
           1, SENSORS_TEMPERATURE_HANDLE,
-          SENSOR_TYPE_TEMPERATURE, 85.0f, CONVERT_TEMPERATURE, 0.35f, 0, { } },
+          SENSOR_TYPE_TEMPERATURE, 85.0f, CONVERT_TEMPERATURE, 0.35f, 0, 0, 0, { } },
         { "ISL29023 Light sensor",
           "Intersil",
           1, SENSORS_LIGHT_HANDLE,
-          SENSOR_TYPE_LIGHT, 16000.0f, 1.0f, 0.35f, 0, { } },
+          SENSOR_TYPE_LIGHT, 16000.0f, 1.0f, 0.35f, 0, 0, 0, { } },
 };
 
 
diff --git a/mx6/libcamera2/CameraModule.cpp b/mx6/libcamera2/CameraModule.cpp
index 70af149..d7631e4 100755
--- a/mx6/libcamera2/CameraModule.cpp
+++ b/mx6/libcamera2/CameraModule.cpp
@@ -59,6 +59,8 @@ camera_module_t HAL_MODULE_INFO_SYM = {
     get_number_of_cameras: camera_get_number_of_cameras,
     get_camera_info: camera_get_camera_info,
     set_callbacks: NULL,
+    get_vendor_tag_ops: NULL,
+    reserved: {0}
 };
 
 typedef struct fsl_camera_device {
@@ -482,7 +484,7 @@ static void GetCameraPropery(char *pFaceBackCameraName,
                              int  *pFaceBackOrient,
                              int  *pFaceFrontOrient)
 {
-    char orientStr[10];
+    char orientStr[92];
 
     property_get(FACE_BACK_CAMERA_NAME,
                  pFaceBackCameraName,
diff --git a/mx6/libcamera2/CameraUtil.h b/mx6/libcamera2/CameraUtil.h
index 639c375..4c68ca7 100755
--- a/mx6/libcamera2/CameraUtil.h
+++ b/mx6/libcamera2/CameraUtil.h
@@ -83,7 +83,7 @@ using namespace android;
 #define NUM_CAPTURE_BUFFER      1
 
 #define CAMAERA_FILENAME_LENGTH 256
-#define CAMERA_SENSOR_LENGTH    32
+#define CAMERA_SENSOR_LENGTH    92
 #define CAMERA_FORMAT_LENGTH    32
 #define CAMER_PARAM_BUFFER_SIZE 512
 #define PARAMS_DELIMITER ","
diff --git a/mx6/libcamera2/MetadaManager.cpp b/mx6/libcamera2/MetadaManager.cpp
index 4d2508d..76176dc 100755
--- a/mx6/libcamera2/MetadaManager.cpp
+++ b/mx6/libcamera2/MetadaManager.cpp
@@ -667,12 +667,12 @@ status_t MetadaManager::createStaticInfo(camera_metadata_t **info, bool sizeRequ
     static const int32_t lensShadingMapSize[] = {1, 1};
     ADD_OR_SIZE(ANDROID_LENS_INFO_SHADING_MAP_SIZE, lensShadingMapSize,
             sizeof(lensShadingMapSize)/sizeof(int32_t));
-
+#if 0
     static const float lensShadingMap[3 * 1 * 1 ] =
             { 1.f, 1.f, 1.f };
     ADD_OR_SIZE(ANDROID_LENS_INFO_SHADING_MAP, lensShadingMap,
             sizeof(lensShadingMap)/sizeof(float));
-
+#endif
     int32_t lensFacing = mCameraId ?
             ANDROID_LENS_FACING_FRONT : ANDROID_LENS_FACING_BACK;
     ADD_OR_SIZE(ANDROID_LENS_FACING, &lensFacing, 1);
diff --git a/mx6/libcamera2/StreamAdapter.cpp b/mx6/libcamera2/StreamAdapter.cpp
index 58cd23b..188c6d8 100755
--- a/mx6/libcamera2/StreamAdapter.cpp
+++ b/mx6/libcamera2/StreamAdapter.cpp
@@ -67,7 +67,7 @@ int StreamAdapter::start()
     mTime1 = mTime2 = 0;
     mTotalFrames = mFps = 0;
     mShowFps = false;
-    char prop_value[CAMERA_FORMAT_LENGTH];
+    char prop_value[CAMERA_SENSOR_LENGTH];
     if (property_get("sys.camera.fps", prop_value, "0")) {
         if (strcmp(prop_value, "1") == 0) {
             mShowFps = true;
-- 
1.8.0

